File: src/main/java/com/tribalscale/test/service/AccountTransferService.java
package com.tribalscale.test.service;

import com.tribalscale.test.domain.Account;
import com.tribalscale.test.domain.MoneyTransfer;
import reactor.core.publisher.Mono;

public interface AccountTransferService {
    Mono<Integer> transferMoneyFromAccountToAnotherAccount(MoneyTransfer moneyTransfer);

    Mono<Account> getAccountDetailsByAccountId(String accountId);
}

File: src/main/java/com/tribalscale/test/service/AccountTransferService.kt
package com.tribalscale.test.service

import com.tribalscale.test.domain.Account
import com.tribalscale.test.domain.MoneyTransfer
import reactor.core.publisher.Mono

interface AccountTransferService {
    fun transferMoneyFromAccountToAnotherAccount(moneyTransfer: MoneyTransfer): Mono<Int>
    fun getAccountDetailsByAccountId(accountId: String): Mono<Account>
}